service/src/main/java/uk/nhs/adaptors/gp2gp/mhs/MhsRequestBuilder.java

```diff
@@ -0,0 +1,93 @@
+import org.springframework.web.reactive.function.BodyInserter;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.netty.http.client.HttpClient;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDefinition;
+import uk.nhs.adaptors.gp2gp.ehr.SendEhrExtractCoreTaskDefinition;
+import org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
+public class MhsRequestBuilder {
+    private static final String INTERACTION_ID = "Interaction-Id";
+    private static final String JSON_CONTENT_TYPE = "application/json";
+    private static final String MHS_OUTBOUND_INTERACTION_ID = "RCMR_IN030000UK06";
+    private static final int BYTE_COUNT = 16 * 1024 * 1024;
+
+    private final InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream("StubEhrExtractCoreMessage.json");
+    private final String stubExtractCoreMessage = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+    private final MhsConfiguration mhsConfiguration;
+    private final MhsWebClientFilter mhsWebClientFilter;
+
+    public RequestHeadersSpec<?> buildSendEhrExtractCoreRequest(SendEhrExtractCoreTaskDefinition sendEhrExtractCoreTaskDefinition) {
+        SslContext sslContext = buildSSLContext();
+        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+        WebClient client = buildWebClient(httpClient);
+
+        WebClient.RequestBodySpec uri = client
+            .method(HttpMethod.POST)
+            .uri(mhsConfiguration.getUrl());
+
+        BodyInserter<Object, ReactiveHttpOutputMessage> bodyInserter = BodyInserters.fromValue(stubExtractCoreMessage);
+
+        return buildRequestWithHeadersAndBody(uri, stubExtractCoreMessage, bodyInserter,
+            sendEhrExtractCoreTaskDefinition, MHS_OUTBOUND_INTERACTION_ID);
+    }
+
+    @SneakyThrows
+    private SslContext buildSSLContext() {
+        return SslContextBuilder
+            .forClient()
+            .trustManager(InsecureTrustManagerFactory.INSTANCE)
+            .build();
+    }
+
+    private WebClient buildWebClient(HttpClient httpClient) {
+        return WebClient
+            .builder()
+            .exchangeStrategies(buildExchangeStrategies())
+            .clientConnector(new ReactorClientHttpConnector(httpClient))
+            .filter(mhsWebClientFilter.errorHandlingFilter())
+            .baseUrl(mhsConfiguration.getUrl())
+            .defaultUriVariables(Collections.singletonMap("url", mhsConfiguration.getUrl()))
+            .build();
+    }
+
+    private RequestHeadersSpec<?>  buildRequestWithHeadersAndBody(WebClient.RequestBodySpec uri, String requestBody,
```

Comment:
requestBody and task definition is not used in the method. I am not sure we need this method maybe it would be enough to build all uri in 49 line with other params and to not split it into 2 methods. What do you think?